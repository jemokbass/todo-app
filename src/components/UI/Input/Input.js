import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';

const Input = forwardRef((props, ref) => {
  let {
    type,
    label,
    className,
    autoComplete,
    autoCorrect,
    name,
    errors,
    errorsMessage,
    onChange,
    onBlur,
    placeholder,
  } = props;
  className = className ? ` ${className}` : '';

  return (
    <label className="label input-field">
      <span className="input-field__label">{label}</span>
      <input
        className={`input${className}`}
        type={type}
        autoComplete={autoComplete}
        autoCorrect={autoCorrect}
        name={name}
        ref={ref}
        onChange={onChange}
        onBlur={onBlur}
        placeholder={placeholder}
      />
      {errors && <span className="input-field__error">{errorsMessage}</span>}
    </label>
  );
});

Input.defaultProps = {
  label: '',
  type: 'text',
  onChange: () => {},
  value: '',
  autoComplete: 'off',
  autoCorrect: 'off',
};

Input.propTypes = {
  label: PropTypes.string,
  type: PropTypes.oneOf(['text', 'radio', 'checkbox', 'email', 'file', 'number', 'tel', 'password']),
  onChange: PropTypes.func,
  value: PropTypes.string,
  autoComplete: PropTypes.oneOf(['off', 'on']),
  autoCorrect: PropTypes.oneOf(['off', 'on']),
};

export default Input;
