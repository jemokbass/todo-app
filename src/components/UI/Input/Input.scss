.input-field {
  display: block;

  &__label {
    display: block;
    font-size: 22px;
  }

  .input {
  }
}
