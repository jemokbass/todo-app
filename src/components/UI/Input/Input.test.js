import React from 'react';
import { render, screen } from '@testing-library/react';

import Input from './Input';

describe('Input UI component', () => {
  it('Should render input component', () => {
    render(<Input />);
  });

  it('Should render input with placeholder', () => {
    render(<Input placeholder="placeholder" />);
    expect(screen.getByPlaceholderText(/placeholder/i)).toBeInTheDocument();
  });

  it('Should render input with name', () => {
    render(<Input name="name" />);
  });

  describe('Should render snapshots', () => {
    it('Should render empty input snapshot', () => {
      const { asFragment } = render(<Input />);
      expect(asFragment(<Input />)).toMatchSnapshot();
    });

    it('Should render input with label snapshot', () => {
      const { asFragment } = render(<Input label="label" />);
      expect(asFragment(<Input label="label" />)).toMatchSnapshot();
    });

    it('Should render disabled input snapshot', () => {
      const { asFragment } = render(<Input disabled />);
      expect(asFragment(<Input disabled />)).toMatchSnapshot();
    });
  });

  describe('Default props', () => {
    it('Should use default label', () => {
      const res = Input.defaultProps.label;
      expect(res).toBe('');
    });

    it('Should use default type', () => {
      const res = Input.defaultProps.type;
      expect(res).toBe('text');
    });

    it('Should use default autoComplete', () => {
      const res = Input.defaultProps.autoComplete;
      expect(res).toBe('off');
    });

    it('Should use default autoCorrect', () => {
      const res = Input.defaultProps.autoCorrect;
      expect(res).toBe('off');
    });

    it('Should use default disabled', () => {
      const res = Input.defaultProps.disabled;
      expect(res).toBe(false);
    });

    it('Should use default name', () => {
      const res = Input.defaultProps.name;
      expect(res).toBe('');
    });
  });
});
