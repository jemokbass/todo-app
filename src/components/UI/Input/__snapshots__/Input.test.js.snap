// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input UI component Should render snapshots Should render disabled input snapshot 1`] = `
<DocumentFragment>
  <label
    class="label input-field"
  >
    <span
      class="input-field__label"
    />
    <input
      autocomplete="off"
      autocorrect="off"
      class="input"
      disabled=""
      name=""
      type="text"
    />
  </label>
</DocumentFragment>
`;

exports[`Input UI component Should render snapshots Should render empty input snapshot 1`] = `
<DocumentFragment>
  <label
    class="label input-field"
  >
    <span
      class="input-field__label"
    />
    <input
      autocomplete="off"
      autocorrect="off"
      class="input"
      name=""
      type="text"
    />
  </label>
</DocumentFragment>
`;

exports[`Input UI component Should render snapshots Should render input with label snapshot 1`] = `
<DocumentFragment>
  <label
    class="label input-field"
  >
    <span
      class="input-field__label"
    >
      label
    </span>
    <input
      autocomplete="off"
      autocorrect="off"
      class="input"
      name=""
      type="text"
    />
  </label>
</DocumentFragment>
`;
